packages/kanel-enum-tables/src/generateEnumTypesFromTableValuesModule.ts

```diff
@@ -0,0 +1,149 @@
+import type {
+  GenericDeclaration,
+  InterfaceDeclaration,
+  Output,
+  PreRenderHook,
+  TypeDeclaration,
+} from 'kanel';
+import { escapeName } from 'kanel';
+import knex from 'knex';
+
+const generateEnumTypesFromTableValuesModule: PreRenderHook = async (
+  outputAccumulator,
+  instantiatedConfig
+) => {
+  const connection = instantiatedConfig.connection;
+  const db = knex({ client: 'postgres', connection });
+
+  // Get all the tables in all the schemas
+  const allTables = Object.values(instantiatedConfig.schemas)
+    .map(({ tables }) => tables)
+    .flat();
+
+  // ..and find the ones that have an "@enum" comment.
+  const enumTables = allTables.filter(
+    (table) => table.comment !== null && table.comment.indexOf('@enum') !== -1
+  );
+
+  const overrides: Output = {};
+
+  for (const table of enumTables) {
+    const primaryKeyColumns = table.columns.filter(
+      (column) => column.isPrimaryKey
+    );
+
+    if (primaryKeyColumns.length !== 1) {
+      throw new Error(
+        `Enum table ${table.schemaName}.${table.name} must have exactly one primary key column`
+      );
+    }
+
+    const primaryKeyColumn = primaryKeyColumns[0];
+
+    const primaryKeyTypeDeclaration = instantiatedConfig.generateIdentifierType(
+      primaryKeyColumn,
+      table,
+      instantiatedConfig
+    );
+
+    const selectorMetadata = instantiatedConfig.getMetadata(
+      table,
+      'selector',
+      instantiatedConfig
+    );
+
+    const initializerMetadata = instantiatedConfig.getMetadata(
+      table,
+      'initializer',
+      instantiatedConfig
+    );
+
+    const mutatorMetadata = instantiatedConfig.getMetadata(
+      table,
+      'mutator',
+      instantiatedConfig
+    );
+
+    const declarations =
+      outputAccumulator[selectorMetadata.path]?.declarations ?? [];
+
+    // TODO: Get @enumName and use it as the name of the enum type
+    // TODO: Get @enumDescription (default "descritpion") and use it as the comment for the enum type
+
+    const rows = await db
+      .withSchema(table.schemaName)
+      .select(primaryKeyColumn.name)
+      .from(table.name)
+      .orderBy(primaryKeyColumn.name);
+
+    const newDeclarations = declarations.map((declaration) => {
+      if (
+        declaration.declarationType === 'typeDeclaration' &&
+        declaration.name === primaryKeyTypeDeclaration.name
+      ) {
+        if (instantiatedConfig.enumStyle === 'type') {
+          const newDeclaration: TypeDeclaration = {
+            ...declaration,
+            typeDefinition: [
+              '', // Start definition on new line
+              ...rows.map((row) => `| '${row.name}'`),
+              `& { __flavor?: '${declaration.name}' }`,
```

Comment:
Do we need to use getMetadata or something to make sure the type definition is configurable enough? What if someone is using `__brand`?

Reply:
That's a good question. I don't have a great strategy for extensions like this yet. There are also a good number of things that needs to be configurable in the Zon extension and it's hard to do it in a simple way that still works with the many possible configurations. 

For this specific instance, why are you adding a flavor to a union type? Is that to avoid overlapping types?

Reply:
So I probably just added if becuase of some copy-paste to be honest.

I did a copy-paste of the kanel-knex plugin and then probably copied some things from my own getMetadata function that uses ´__flavor?: ...` and didn't really think it through.

Just had a look in the makeEnumsGenerator and that there is no brand/flavour for the enum type defs and this should just be removed from here.